pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -679,59 +848,58 @@ func (p *MasqueradePodInterface) createNatRulesUsingNftables() error {
 }
 
 type SlirpPodInterface struct {
-	vmi             *v1.VirtualMachineInstance
-	iface           *v1.Interface
-	domain          *api.Domain
-	podInterfaceNum int
+	vmi       *v1.VirtualMachineInstance
+	iface     *v1.Interface
+	virtIface *api.Interface
+	domain    *api.Domain
 }
 
 func (s *SlirpPodInterface) discoverPodNetworkInterface() error {
-	s.domain.Spec.QEMUCmd.QEMUArg = append(s.domain.Spec.QEMUCmd.QEMUArg, api.Arg{Value: "-device"})
 	return nil
 }
 
 func (s *SlirpPodInterface) preparePodNetworkInterfaces() error {
-	interfaces := s.domain.Spec.Devices.Interfaces
-	domainInterface := interfaces[s.podInterfaceNum]
-	s.domain.Spec.QEMUCmd.QEMUArg = append(s.domain.Spec.QEMUCmd.QEMUArg, api.Arg{Value: fmt.Sprintf("%s,netdev=%s", domainInterface.Model.Type, s.iface.Name)})
-
-	s.domain.Spec.Devices.Interfaces = append(interfaces[:s.podInterfaceNum], interfaces[s.podInterfaceNum+1:]...)
-	s.podInterfaceNum = len(s.domain.Spec.QEMUCmd.QEMUArg) - 1
+	return nil
+}
 
+func (s *SlirpPodInterface) startDHCP(vmi *v1.VirtualMachineInstance) error {
 	return nil
 }
 
 func (s *SlirpPodInterface) decorateConfig() error {
-	s.domain.Spec.QEMUCmd.QEMUArg[s.podInterfaceNum].Value += fmt.Sprintf(",id=%s", s.iface.Name)
+	// remove slirp interface from domain spec devices interfaces
+	ifaces := s.domain.Spec.Devices.Interfaces
+	for i, iface := range ifaces {
+		if iface.Alias.Name == s.iface.Name {
+			s.domain.Spec.Devices.Interfaces = append(ifaces[:i], ifaces[i+1:]...)
+			break
+		}
+	}
+
+	qemuArg := fmt.Sprintf("%s,netdev=%s,id=%s", api.GetInterfaceType(s.iface), s.iface.Name, s.iface.Name)
```

Comment:
just a nit, but perhaps we don't need another call to `api.GetInterfaceType` as this has to be already set in `interface.Model.Type` - since the convertor has already stored it there.
we can just get the `interface` from the above loop